src/block/parser.rs

```diff
@@ -100,7 +108,21 @@ impl BlockParser {
                     paths.len(),
                     display_duration(time.elapsed()),
                 );
-                info!("Searching for canonical chain in startup blocks");
+
+                if let Some(blockchain_length) = length_filter {
```

Comment:
why do we need this?

Reply:
If I run `mainnet-test` on a block dir with 275k blocks, I don't want to have to wait for canonical chain discovery to then just apply 10k blocks

Reply:
So this effectively makes it so that you only consider the first `blockchain_length` blocks in the dir